File: src/main/java/net/igneo/icv/networking/packet/EntitySyncS2CPacket.java
package net.igneo.icv.networking.packet;

import net.igneo.icv.init.ICVUtils;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraftforge.network.NetworkEvent;

import java.util.function.Supplier;


public class EntitySyncS2CPacket {
    private final int entityID;
    private final int slot;
    
    public EntitySyncS2CPacket(int entityID, int slot) {
        this.entityID = entityID;
        this.slot = slot;
    }
    
    public EntitySyncS2CPacket(FriendlyByteBuf buf) {
        this.entityID = buf.readInt();
        this.slot = buf.readInt();
    }
    
    public void toBytes(FriendlyByteBuf buf) {
        buf.writeInt(entityID);
        buf.writeInt(slot);
    }
    
    public boolean handle(Supplier<NetworkEvent.Context> supplier) {
        NetworkEvent.Context context = supplier.get();
        context.enqueueWork(() -> {
            System.out.println("syncing");
            ICVUtils.syncClientEntity(entityID, slot);
        });
        return true;
    }
}

File: src/main/java/net/igneo/icv/networking/packet/EntitySyncS2CPacket.kt
package net.igneo.icv.networking.packet

import net.igneo.icv.Utils.syncClientEntity
import net.minecraft.network.FriendlyByteBuf
import net.minecraftforge.network.NetworkEvent
import java.util.function.Supplier

class EntitySyncS2CPacket {
    private val entityID: Int
    private val slot: Int

    constructor(entityID: Int, slot: Int) {
        this.entityID = entityID
        this.slot = slot
    }

    constructor(buf: FriendlyByteBuf) {
        this.entityID = buf.readInt()
        this.slot = buf.readInt()
    }

    fun toBytes(buf: FriendlyByteBuf) {
        buf.writeInt(entityID)
        buf.writeInt(slot)
    }

    fun handle(supplier: Supplier<NetworkEvent.Context>): Boolean {
        val context = supplier.get()
        context.enqueueWork {
            println("syncing")
            syncClientEntity(entityID, slot)
        }
        return true
    }
}
